dlib/vectorstream/vectorstream.h

```diff
@@ -43,6 +42,32 @@ namespace dlib
             {
                 read_pos = pos;
             }
+            
+            pos_type seekpos(pos_type pos, std::ios_base::openmode mode = std::ios_base::in | std::ios_base::out)
+            {
+                return seekoff(pos - pos_type(off_type(0)), std::ios_base::beg, mode);
+            }
+
+            pos_type seekoff(off_type off, std::ios_base::seekdir dir,
+                             std::ios_base::openmode mode = std::ios_base::in | std::ios_base::out )
+            {
+                (void)mode; //don't care what mode is. This class handles both
```

Comment:
`mode` can't be ignored.  `mode` might be indicating that only the write pointer is moved, which is just not something the vectorstream supports.  You at least need a `DLIB_ASSERT` on this to claim that `mode` is `in`.

Reply:
Yep agreed